base/loading.jl

```diff
@@ -435,23 +412,14 @@ function require(mod::Symbol)
         # just load the file normally via include
         # for unknown dependencies
         try
-            if last && myid() == 1 && nprocs() > 1
-                # include on node 1 first to check for PrecompilableErrors
-                eval(Main, :(Base.include_from_node1($path)))
-
-                # broadcast top-level import/using from node 1 (only)
-                refs = Any[ @spawnat p eval(Main, :(Base.include_from_node1($path))) for p in filter(x -> x != 1, procs()) ]
-                for r in refs; wait(r); end
-            else
-                eval(Main, :(Base.include_from_node1($path)))
-            end
+            eval(Main, :(Base.include_from_node1($path)))
```

Comment:
This risks making the node module state inconsistent since the behavior of `__precompile__` is not uniform across nodes, resulting in a race condition that this block of code previously tried to heuristically avoid. To solve, I think we need to remove the `myid` test from `__precompile__`, but also to make it possible to ignore a `__precompile__` statement (perhaps some global dictionary of `sourcepath` names for which `__precompile__` is temporarily disabled?). Then when we catch a `precompilableerror`, we can first request the cache compile file from node 1 (the `package_locks` synchronization might be able to help synchronize these requests). Once we know if that will succeed for fail, we can proceed with either loading the cache file, or include it with `__precompile__` disabled for that file. (this should also be changed to be the case on node 1, not just worker nodes, as I think this logic should be more robust – ignoring the cache whenever it is unusable – but just hadn't been implemented yet)